### PR TITLE
Identifier: Sanitize against e.g. multi-line strings

### PR DESCRIPTION
**Pull request by sschuberth**
/cc @zenogantner

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/660)

### CODE REVIEW

File: model/src/main/kotlin/Identifier.kt [hunk 1]
```diff
@@ -80,8 +80,10 @@ data class Identifier(
         }
     }
 
+    private val list = listOf(provider, namespace, name, version)
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/Identifier.kt]: `list` is so generic, what about `components`?

File: model/src/main/kotlin/Identifier.kt [hunk 2]
```diff
@@ -122,10 +124,10 @@ data class Identifier(
     /**
      * Create a path based on the properties of the [Identifier]. All properties are encoded using [encodeOrUnknown].
      */
-    fun toPath() = listOf(provider, namespace, name, version).joinToString("/") { it.encodeOrUnknown() }
+    fun toPath() = list.joinToString("/") { it.encodeOrUnknown() }
 
     // TODO: Consider using a PURL here, see https://github.com/package-url/purl-spec#purl.
-    override fun toString() = "$provider:$namespace:$name:$version"
+    override fun toString() = list.joinToString(":") { it.trim().filterNot { it < ' ' } }
```
Comment (mnonnenmacher) [inline model/src/main/kotlin/Identifier.kt]: Is it sufficient to do this here? We probably print components of the identifier in other places, too, so doing it in the constructor already would make sure it is always clean.

Comment (sschuberth) [reply model/src/main/kotlin/Identifier.kt]: True. I was hoping I could use property delegation with something like a `TrimmedSingleLine` class, but unfortunately [it's not possible to use delegated properties in the primary constructor of data classes](https://discuss.kotlinlang.org/t/using-delegated-properties-in-data-classes/6858).

File: model/src/main/kotlin/Identifier.kt (hunk 1 above)
Comment (sschuberth) [reply model/src/main/kotlin/Identifier.kt]: I was choosing `list` as that#s the name we're already using in `fromString()`, but I'll change both.

**Pull request merged**